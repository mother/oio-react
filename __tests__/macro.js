import pluginTester from 'babel-plugin-tester'
import plugin from 'babel-plugin-macros'

pluginTester({
   plugin,
   babelOptions: { filename: __filename },
   snapshot: true,
   tests: {
      'works with ordinary string literals': {
         code: `
            import OIOResponsiveStringMacro from '../macro'
            const result = OIOResponsiveStringMacro('100[a] 200[b]')`
      }
      // 'works with template literals': {
      //    code: "import OIOResponsiveStringMacro from '../macro' \n" +
      //       "const result = OIOResponsiveStringMacro(`${a}[a] 100[b]`)"
      // }
   }
})
