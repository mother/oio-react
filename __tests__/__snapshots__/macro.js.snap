// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros works with ordinary string literals: works with ordinary string literals 1`] = `
"
import OIOResponsiveStringMacro from '../macro'
const result = OIOResponsiveStringMacro('100[a] 200[b]')

      ↓ ↓ ↓ ↓ ↓ ↓

const result = {
  breakpointsWereSet: true,
  a: \\"100\\",
  b: \\"200\\"
};
"
`;
